### PR TITLE
feat: add exp_cronbach_alpha reliability analysis (#14309)

### PR DESCRIPTION
**Pull request by hidekoji**
## Summary

Adds `exp_cronbach_alpha()` — Cronbach's Alpha / reliability analysis — powering the new **Cronbach's Alpha** Analytics type in the desktop app (tam PR to follow).

Supports **Pearson** (Cronbach's Alpha), **Polychoric** (Ordinal Alpha), and **Mixed Correlation** (Mixed-Correlation Alpha), with an `auto` selector that picks the method from the item types. Reuses `psych` (already an Imports dep — no new dependency).

## Contract

- `exp_cronbach_alpha(df, ..., correlation_method = "auto", check_keys = FALSE, smooth = TRUE, correct = 0.5, max_nrow = NULL, seed = 1)` — returns a `model_df` (class `cronbach_alpha_exploratory`), one model per group.
- `glance(model)` — overall alpha, standardized alpha, CI, G6, counts, average inter-item correlation, correlation type, interpretation.
- `tidy(model, type = ...)` — `summary` / `alpha_if_dropped` / `item_stats` / `correlation` / `response_distribution` / `scale_candidates` / `data`.

## Testing

`tests/testthat/test_reliability.R` covers the Pearson, all-ordinal, mixed, reverse-key, and edge-case contracts:
- Pearson alpha matches `psych::alpha`'s `raw_alpha` exactly; every section shape verified.
- Auto→Polychoric (ordered factors) → Ordinal Alpha; `raw.r` hidden.
- Mixed correlation → Mixed-Correlation Alpha; CI and `raw.r` are hidden by specification.
- <2 variables error; nominal-factor error; constant-column drop; `group_by`.

## Notes

- `NAMESPACE` updated with the 3 new exports (`exp_cronbach_alpha`, `tidy`/`glance` S3 methods). `devtools::document()` mis-parses in this local env due to an unrelated `find_data` load error, so the entries were inserted by hand exactly as document() would emit them — please regenerate in CI if preferred.
- `DESCRIPTION` version bumped 16.0.5 → 16.0.6. The bundled binary must be rebuilt + the tam `requiredRPackagesReduced.json` bumped before the desktop feature works end-to-end.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Specification alignment follow-up

- Pearson exposes Feldt/Duhachek 95% CI and `raw.r`.
- Polychoric/Mixed omit the CI summary row and the `raw.r` output column.
- `check.keys` is applied consistently to matrix statistics, item statistics, and alpha-if-dropped results.
- Local full test execution still requires the repository's 4.6 `devtools`/`exploratory` dependency stack.

### CODE REVIEW

File: DESCRIPTION [hunk 1]
```diff
@@ -1,7 +1,7 @@
 Package: exploratory
 Type: Package
 Title: R package for Exploratory
-Version: 16.0.5
+Version: 16.0.6
```
Comment (hidekoji) [inline DESCRIPTION]:
```suggestion
Version: 16.0.5
```

**Review by kei51e — APPROVED** [review]

**Comment by hidekoji** [comment]
Follow-up for the Cronbach's Alpha specification alignment:

- Updated `reliability_response_distribution()` to match Summary View numeric behavior: low-cardinality integer values remain discrete; other numeric values use ten equal-width bins.
- Added regression coverage for low-cardinality, continuous, and all-missing numeric columns.
- R source and test files pass syntax parsing. Full test execution is still blocked in this local environment because `dplyr` and `testthat` are not installed.

Commit: `fb9875b8`

**Comment by hidekoji** [comment]
Removed the unrelated `Config/roxygen2/version: 8.0.0` metadata change and restored `RoxygenNote: 7.3.3` to match `master`. The roxygen metadata change is not required for `exp_cronbach_alpha()` runtime behavior. Commit: `8140b5b7`.

**Comment by hidekoji** [comment]
Follow-up: restored the original `DESCRIPTION` field order as well, so the PR no longer carries a `DESCRIPTION` diff relative to `master`. Commit: `b2ece06f`.
